packages/client/src/pages/register.vue

```diff
@@ -1,146 +1,233 @@
+          {{ t("auth.alreadyRegistered") }}
+        </span>
+
+        <q-btn
+          :label="t('auth.goToLogin')"
+          class="full-width outline-black-12"
+          outline
+          to="login"
+        />
+      </q-card-section>
+
+      <q-separator />
+
+      <router-link
+        :to="{ name: AvailableRouteNames.ForgotPassword }"
+        class="text-black-87 text-center text-subtitle1"
+        data-cy="forgot-password-link"
+      >
+        {{ t("auth.forgotPassword") }}
+      </router-link>
+    </q-card>
   </q-page>
 </template>
 
 <script setup lang="ts">
 import { ApolloError } from "@apollo/client/core";
-import { reactive, ref, toRaw } from "vue";
+import {
+  mdiEye,
+  mdiEyeOff,
+  mdiInformationOutline,
+} from "@quasar/extras/mdi-v7";
+import { QInputProps } from "quasar";
+import { computed, reactive, ref, toRaw } from "vue";
 import { useI18n } from "vue-i18n";
-import PasswordStrengthBar from "components/password-strength-bar.vue";
 import SocialAuthButtons from "components/social-auth-buttons.vue";
-import { RegisterPayload } from "src/@generated/graphql";
-import KPasswordInput from "src/components/k-password-input.vue";
-import { STRENGTH_BAR_STEPS } from "src/components/models";
+import { RegisterUserPayload } from "src/@generated/graphql";
+import { useTheme } from "src/composables/use-theme";
 import { notifyError } from "src/helpers/error-messages";
 import {
-  emailRule,
   makeValueMatchRule,
   requiredRule,
   validatePasswordRule,
 } from "src/helpers/rules";
+import { AvailableRouteNames } from "src/models/routes";
 import { useRegisterMutation } from "src/services/auth";
 
 const SOCIAL_LOGIN_ENABLED = process.env.SOCIAL_LOGIN_ENABLED === "true";
 
 const { t } = useI18n();
 
+const { theme } = useTheme();
+
 const showPassword = ref(false);
 
-const user = reactive<RegisterPayload>({
+type UserRegistrationData = Omit<
+  RegisterUserPayload,
+  "discount" | "retailLocationId" | "notes" | "dateOfBirth"
+> & {
+  confirmEmail: string;
+  dateOfBirth: string;
+};
+
+const user = reactive<UserRegistrationData>({
   email: "",
   firstname: "",
   lastname: "",
   password: "",
+  confirmEmail: "",
+  phoneNumber: "",
   passwordConfirmation: "",
+  dateOfBirth: "",
 });
 
 const { register, loading: isRegistering } = useRegisterMutation();
 
+const emailMatchRule = makeValueMatchRule(
+  () => user.confirmEmail,
+  () => t("auth.emailsDoNotMatch"),
+);
 const passwordMatchRule = makeValueMatchRule(
   () => user.password,
   () => t("auth.passwordDoNotMatch"),
 );
 
+const formData = computed<
+  ({ field: keyof UserRegistrationData; tooltip?: string } & Omit<
+    QInputProps,
+    "modelValue"
+  >)[]
+>(() => [
+  {
+    field: "firstname",
+    label: t("auth.firstName"),
+    rules: [requiredRule],
+  },
+  {
+    field: "lastname",
+    label: t("auth.lastName"),
+    rules: [requiredRule],
+  },
+  {
+    field: "dateOfBirth",
+    label: t("auth.birthDate"),
+    type: "date",
```

Comment:
This one here needs to have the required rule, because otherwise we cannot verify the `delegate`